crates/solvers/src/tests/dex/partial_fill.rs

```diff
@@ -94,7 +94,30 @@ async fn tested_amounts_adjust_depending_on_response() {
     ])
     .await;
 
-    let engine = tests::SolverEngine::new("balancer", balancer::config(&api.address)).await;
+    let simulation_node = mock::http::setup(vec![mock::http::Expectation::Post {
+        path: mock::http::Path::Any,
+        req: mock::http::RequestBody::Any,
+        res: {
+            json!({
+                "id": 1,
+                "jsonrpc": "2.0",
+                "result": "0x0000000000000000000000000000000000000000000000000000000000015B3C"
```

Comment:
I had to use the original gas amount returned by the API to make the test pass. Otherwise the fee amount would have been different. I think that means the code is working.

Reply:
You should also test returning `0` to check the "I can't simulate" logic.